### ISSUE TITLE
[AddSelect]: Nested content ("View children" button) cannot be focused via keyboard

### ISSUE THREAD

**Issue by ariellalgilmore** [carbon-design-system/ibm-products#5451]
### Package

Carbon for IBM Products

### Description

When using keyboard to navigate the addselect components, it is not clear how to reach the view children button. I realized they both navigate with the right arrow key, but you never reach the tooltip/hover state like with mouse

https://github.com/carbon-design-system/ibm-products/assets/20210594/0fca5914-f05c-4817-8520-f77928960a19

<img width="1728" alt="Screenshot 2024-06-10 at 12 46 19 PM" src="https://github.com/carbon-design-system/ibm-products/assets/20210594/755b43d1-83a3-4abd-982f-ff12aae91861">

### Component(s) impacted

singleaddselect
multiaddselect

### @carbon/ibm-products (previously @carbon/ibm-cloud-cognitive) version

2.41.0

### Severity

Severity 3 = The problem is visible or noticeable to users but does not impede the usability or functionality. Affects minor functionality, has a workaround.

### Product/offering

n/a

### CodeSandbox or Stackblitz example

https://carbon-for-ibm-products.netlify.app/?path=/story/ibm-products-patterns-add-and-select-multiaddselect--with-hierarchy&globals=viewport:basic

### Steps to reproduce the issue (if applicable)

go to: https://carbon-for-ibm-products.netlify.app/?path=/story/ibm-products-patterns-add-and-select-multiaddselect--with-hierarchy&globals=viewport:basic

use your keyboard to try to navigate to the caret icon

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate issues

**Comment by sangeethababu9223** [comment]
Hi @janhassel ,
In this issue for both `SingleAddSelect` and `MultiAddSelect` tab key focuses on the first radio item in the group and for navigating through the rest we use `up/down` navigation keys. For items with child elements an icon button(with `aria-hidden` and `tabindex= -1` ,) is shown. We can navigate to the child elements using `right` arrow key.

Do we need to make the icon button tabbable here? Could you please share your suggestion on this?

**Comment by janhassel** [comment]
@sangeethababu9223 The icon button is there for mouse interaction and does not need to be reachable via keyboard as this component follows the treegrid semantics and its respective keyboard interactions.

I did notice however, that in the linked demo, the ArrowLeft interaction doesn't work and sometimes the ArrowUp/ArrowDown breaks as well on the second level.

**Comment by sangeethababu9223** [comment]
Has opened a new issue including the issues mentioned in the above comment.
https://github.com/carbon-design-system/ibm-products/issues/6024

Closing this ticket as tabbing works as expected here.
